app/js/enhancedDate.js

```diff
@@ -1,5 +1,71 @@
+var enhancedDate = (function() {
+	var dateIsSet;
+	var enhancedDateObject = {
+		setDate: setDate,
+		getDate: getDate,
+		getDayName: getDayName,
+		getMonthName: getMonthName,
+		isFuture: isFuture,
+		isToday: isToday
+	};
+	return enhancedDateObject;
+
+
+	function setDate(date) {
+		if (!date) {dateIsSet = new Date(); return;}
+		if (!(date instanceof Date)) {console.warn('sorry, not a date'); return;}
+		dateIsSet = new Date(date);
+	};
+
+	function getDate() {
+		isDateSet();
+		return dateIsSet;
+	}
+
+	function getDayName() {
+		isDateSet();
+		var dayName = dateIsSet.getDay();
+		var dayOfWeekArray = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+		return dayOfWeekArray[dayName];
+	}
+	function isDateSet() {
+		if(!dateIsSet) {console.warn('date hasn\'t been set! Please call setDate first!'); return false;}
+	}
+	function getMonthName() {
+		isDateSet();
+		var monthName = dateIsSet.getMonth();
+		var monthArray = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+		return monthArray[monthName];
+	};
+	function isFuture() {
+		isDateSet();
+		if(dateIsSet.getTime() > new Date().getTime()) {
+			return true;
+		}
+		return false;
+	};
+	function isToday() {
+		isDateSet();
+		if(today(dateIsSet) === today(new Date())) {return true;}
+		return false;
+	}
+	function today(input) {
+		var day = input.getDay();
```

Comment:
Instead of getDay for this comparison, use getDate - you will have multiple days during the same month of the same year that are the same (3 - 4 Mondays, 3 - 4 Tuesdays, etc.)